File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/home/HomePlayer.java
package calebxzhou.rdi.mc.common2.home;

import java.util.Map;

public interface HomePlayer {
    HomeLocation currentLocation();

    Map<String, HomeLocation> loadHomes();

    void saveHomes(Map<String, HomeLocation> homes);

    HomeResult teleportTo(HomeLocation location);
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/home/HomePlayer.kt
package calebxzhou.rdi.mc.rcmd.home

interface HomePlayer {
    fun currentLocation(): HomeLocation

    fun loadHomes(): MutableMap<String, HomeLocation>

    fun saveHomes(homes: MutableMap<String, HomeLocation>)

    fun teleportTo(location: HomeLocation): HomeResult
}
